diff --git a/libs/bluevk/src/BlueVKLinuxAndroid.cpp b/libs/bluevk/src/BlueVKLinuxAndroid.cpp
index dbde8d802..9505c80cd 100644
--- a/libs/bluevk/src/BlueVKLinuxAndroid.cpp
+++ b/libs/bluevk/src/BlueVKLinuxAndroid.cpp
@@ -27,7 +27,7 @@ static void* module = nullptr;
 // Determines the file path to the Vulkan library and calls dlopen on it.
 // Returns false on failure.
 bool loadLibrary() {
-    const char* path = nullptr;
+    const char* path = "./libvk_swiftshader.so";
 
     // For security, consult an environment variable only if Filament is built with this #define
     // enabled. Our weekly GitHub releases do not have this enabled.
diff --git a/libs/bluevk/src/BlueVKWindows.cpp b/libs/bluevk/src/BlueVKWindows.cpp
index b787c742a..cc4c5e5d6 100644
--- a/libs/bluevk/src/BlueVKWindows.cpp
+++ b/libs/bluevk/src/BlueVKWindows.cpp
@@ -19,7 +19,7 @@
 
 namespace bluevk {
 
-static const char* VKLIBRARY_PATH = "vulkan-1.dll";
+static const char* VKLIBRARY_PATH = "vk_swiftshader.dll";
 
 static HMODULE library = nullptr;
 
