diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc252f8a2..d12cb7f97 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -101,27 +101,6 @@ if(CCACHE_FOUND)
     set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK ccache)
 endif()
 
-###########################################################
-# Install Gerrit commit hook
-###########################################################
-
-if(NOT EXISTS ${CMAKE_SOURCE_DIR}/.git/hooks/commit-msg)
-    message(WARNING "
-        .git/hooks/commit-msg was not found.
-        Downloading from https://gerrit-review.googlesource.com/tools/hooks/commit-msg...
-    ")
-
-    file(DOWNLOAD https://gerrit-review.googlesource.com/tools/hooks/commit-msg ${CMAKE_SOURCE_DIR}/commit-msg)
-
-    file(COPY ${CMAKE_SOURCE_DIR}/commit-msg
-         DESTINATION ${CMAKE_SOURCE_DIR}/.git/hooks/
-         FILE_PERMISSIONS
-           OWNER_READ OWNER_WRITE OWNER_EXECUTE
-           GROUP_READ GROUP_WRITE GROUP_EXECUTE
-           WORLD_READ WORLD_EXECUTE)
-    file(REMOVE ${CMAKE_SOURCE_DIR}/commit-msg)
-endif()
-
 ###########################################################
 # Host libraries
 ###########################################################
